### DISCUSSION TITLE
Get specific payment from mercadopago api

### DISCUSSION THREAD

**Discussion by Pacocc97** [mercadopago/sdk-react#51] [Q&A] [answered]
Hello my people.
How do you get the information of a specific payment from mercadopago api?
I'm aware that I need to make a get request from "[https://api.mercadopago.com/v1/payments/{payment](https://api.mercadopago.com/v1/payments/%7Bpayment) id}", but I'm have some trouble fetching from my front end app, due to mercadopago CORS policy!!

Has anyone else had the same problem??

**Comment by ghost** [comment]
Hello, @Pacocc97!
Here is the [documentation](https://www.mercadopago.com.br/developers/en/reference/payments/_payments_id/get) that explains how to retrieve payment data making a direct call to Mercado Pago Payments API.
If you are trying to show the payment status in your frontend app, I suggest you use the Status Screen Brick to facilitate your integration, here its [documentation](https://www.mercadopago.com.br/developers/en/docs/checkout-bricks/status-screen-brick/introduction). You just need to pass your Public Key and the Payment ID, then the magic happens. Status Screen Brick can be integrated with React or vanilla JavaScript.

**Reply by Pacocc97** [reply]
Hello André, thank you for the response. I guess my problem is that I can´t call the payment data from my react app, so to call that API I have to do it from my server ??

**Reply by Pacocc97** [reply]
Because when I fetch with a GET request I get an error.
"Access to XMLHttpRequest at 'https://api.mercadopago.com/v1/payments/NaN' from origin 'https://pruebas.compulan.net.mx' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status."

**Reply by ghost** [reply]
Yes, the call to the API needs to be made on your backend because it requires your Access Token to be passed in the header `Authorization` (check the documentation). Since it's a private key, it cannot be exposed on frontend.
Also note that in the URL you sent https://api.mercadopago.com/v1/payments/NaN, the Payment ID is badly formatted: NaN.
I suggest you to review the code that injects the Payment ID into the API URL.

**Reply by Pacocc97** [reply]
Sorry, my bad. But even if I send the code correctly I still get this error. It´s regarding CORS policy. I´m not sure if its because I'm doing the request from the front end and not the server.
Have you ever made this request from the front end app???

"Access to XMLHttpRequest at 'https://api.mercadopago.com/v1/payments/1315127985' from origin 'http://localhost:3001' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status."

**Reply by ghost** [reply]
No, I have never made the request from frontend since it contains the Access Token. Did you try to make the request from your backend?

**Reply by Pacocc97** [reply]
Not yet. Our rest server is made in Delphi Pascal and there is no SDK for that language, so I´m having a hard time to integrate Mercado Pago to the backend ,

**Reply by ghost** [reply, marked as answer]
Ok, the use of the backend SDKs is not mandatory, you can make direct calls to the Mercado Pago Payments API from your backend since it's public.
As the issue is not related to the SDK React, I recommend you to contact our official developers support: https://www.mercadopago.com.br/developers/pt/support. This link is for Brazil, but you can access the page for your country changing the domain, for example in Argentina it would be: `.com.ar`.

**Reply by Pacocc97** [reply]
Thanks you!! I really appreciate your help!
